comp/forwarder/defaultforwarder/worker.go

```diff
@@ -83,8 +94,24 @@ func NewWorker(
 }
 
 // NewHTTPClient creates a new http.Client
-func NewHTTPClient(config config.Component) *http.Client {
-	transport := httputils.CreateHTTPTransport(config)
+func NewHTTPClient(config config.Component, log log.Component) *http.Client {
+	var transport *http.Transport
+
+	transportConfig := config.Get("forwarder_http_protocol")
+
+	switch transportConfig {
+	case "http1":
+		transport = httputils.CreateHTTPTransport(config, httputils.MaxConnsPerHost(1))
+	case "auto":
+		fallthrough
+	default:
+		if transportConfig != "auto" && log != nil {
+			// The diagnose package calls this function and doesn't have access to a logger,
```

Comment:
Am very open to suggestions on a better way around this!

Reply:
Perhaps diagnose needs to be fixed to provide a valid component or a mock, passing `nil` for a non `Optional` component is not okay.

Reply:
Yes, looking at the code it uses this `http.Client` to send a request to the endpoints to check that the connectivity is Ok. I think it is important that we use the same client to diagnose as we use for the main transactions to ensure we are connecting exactly the same for both.